README.md

```diff
@@ -1 +1,66 @@
 # snare
+
+`snare` is a simple GitHub webhooks runner. When a request comes in, it runs an
+arbitrary program for that repository informing it of the event -- that program
+can then perform whatever actions it wants.
+
+
+## Basic usage
+
+`snare` requires three command-line arguments to be specified:
+
+```
+Usage: snare -p <port> -r <repo scripts dir> -s <secret>
+```
+
+where:
+
+ * `<port>` is a port number (e.g. 4567)
+ * `<repo scripts dir>` is the directory where programs for each repository
+   targeted by the webhook are stored. For a repository `repo` owned by `user`
+   the command `<repo scripts dir>/<user>_<repo> <event> <path to GitHub JSON>`
+   will be run.
+ * `<secret>` is the GitHub secret which guarantees that hooks are coming from
+   your GitHub repository and not a malfeasant.
+
+The most important part of this are the per-repo programs. For example,
+`snare`'s GitHub repository is
+[`https://github.com/softdevteam/snare`](https://github.com/softdevteam/snare):
+if we set up a web hook up that notifies us of when pull request events have
+been made then the command `<repo scripts dir>/softdevteam_snare pull_request
+/path/to/json` will be executed where `pull_request` is the name of the GitHub
+event and `/path/to/JSON` is a path to a file containing the complete GitHub
+JSON for that event. The `softdevteam_snare` program can then execute whatever
+it wants. In order to work out precisely what event has happened, you will
+need to read (GitHub's webhooks
+documentation)[https://developer.github.com/webhooks/].
```

Comment:
I don't understand this paragraph.

> the command `<repo scripts dir>/softdevteam_snare pull_request /path/to/json` will be executed

Sound like you execute a JSON file? That can't be right.

> `/path/to/JSON` is a path to a file containing the complete GitHub JSON for that event.

So you write what github gives you into JSON file? That can't be right either. I must have misunderstood.

Reply:
> the command <repo scripts dir>/softdevteam_snare pull_request /path/to/json will be executed

Wait, I just noticed the spaces. So you do in fact write the JSON to a file, and pass it to a program the user has specified?

Reply:
The script you run will need access to the entire JSON that GitHub sends in order to make decisions based on it. For example, let's say we execute `pr_event.py` whenever a PullRequest event occurs. At this point we don't know what kind of PullRequest event it is. It could be `assigned`, `closed`, `merged`, etc. The script will thus need access to the entire JSON data, which is rather large, in order to do the right thing.

Reply:
OK, that's fine. I just wanted to check I had understood.